Framework/Algorithms/src/Q1DWeighted.cpp

```diff
@@ -129,25 +139,60 @@ void Q1DWeighted::bootstrap(const MatrixWorkspace_const_sptr &inputWS) {
   // number of spectra in the input
   m_nSpec = inputWS->getNumberHistograms();
 
-  // Get wedge properties
-  const int wedges = getProperty("NumberOfWedges");
-  m_nWedges = static_cast<size_t>(wedges);
-  m_wedgeOffset = getProperty("WedgeOffset");
-  m_wedgeAngle = getProperty("WedgeAngle");
-  m_asymmWedges = getProperty("AsymmetricWedges");
-
-  // When symmetric wedges are requested (default), we need to divide
-  // 180/nWedges. When asymmetric wedges are requested, we need to divide
-  // 360/nWedges
-  m_wedgeFullAngle = 180.;
-  if (m_asymmWedges) {
-    m_wedgeFullAngle *= 2;
-  }
-
   // get the number of wavelength bins in the input, note that the input is a
   // histogram
   m_nLambda = inputWS->readY(0).size();
 
+  m_wedgesInnerRadius = std::vector<double>();
+  m_wedgesOuterRadius = std::vector<double>();
+  m_wedgesCenterX = std::vector<double>();
+  m_wedgesCenterY = std::vector<double>();
+  m_wedgesCenterAngle = std::vector<double>();
+  m_wedgesAngleRange = std::vector<double>();
+  m_wedgesIsAsymmetric = std::vector<bool>();
+  m_asymmWedges = getProperty("AsymmetricWedges");
+
+  if (getPropertyValue("ShapeTable").empty()) {
```

Comment:
It is better to check is it is default.
In theory, the workspace might be unnamed, so the property value will be an empty string, but the pointer will be valid.